x/delayedack/ibc_middleware.go

```diff
@@ -148,13 +158,29 @@ func (im IBCMiddleware) OnRecvPacket(
 	rollappPacket := types.RollappPacket{
 		RollappId:   chainID,
 		Packet:      &packet,
-		Status:      types.RollappPacket_PENDING,
+		Status:      commontypes.Status_PENDING,
 		Relayer:     relayer,
 		ProofHeight: ibcClientLatestHeight.GetRevisionHeight(),
 		Type:        types.RollappPacket_ON_RECV,
 	}
 	im.keeper.SetRollappPacket(ctx, rollappPacket)
 
+	// Handle eibc demand order if exists
+	memo := make(map[string]interface{})
+	err = json.Unmarshal([]byte(data.Memo), &memo)
+	if err != nil {
+		logger.Info("Failed to unmarshal memo field", "err", err)
```

Comment:
return nil if `err != nil`
IMO the check is more cleaner (`if err != nil || memo[eibcMemoObjectName] == nil {`)
and log debug here, not info as it will be called for empty memo as well